Updated prospa import for 2d data

tkeller12 opened the pull request:
dims and coords are now determined based on type of experiment. Implemented for 1Pulse2dAverage and T1-IR-FID

tkeller12 committed:
Updated prospa import for 2d data

dims and coords are now determined based on type of experiment. Implemented for 1Pulse2dAverage and T1-IR-FID

ylin00 commented on dnplab/dnpImport/prospa.py:
```diff
@@ -239,3 +228,62 @@ def import_csv(path, return_raw=False, is_complex=True):
+
+        if attrs["delaySpacing"] == "lin":
+            T1 = np.linspace(T1_min_delay, T1_max_delay, T1_steps) / 1000.0
+        else:
+            T1 = np.logspace(T1_min_delay, T1_max_delay, T1_steps) / 1000.0
```
```suggestion
        elif attrs["delaySpacing"] == "log":
            T1 = np.logspace(T1_min_delay, T1_max_delay, T1_steps) / 1000.0
        else:
            raise ValueError(f"Unable to determine delaySpacing {attrs['delaySpacing']}")
```

ylin00 commented on dnplab/dnpImport/prospa.py:
```diff
@@ -239,3 +228,62 @@ def import_csv(path, return_raw=False, is_complex=True):
+    else:
+        dims_list = ["x", "y", "z", "q"]
+        for ix in range(len(data_shape)):
+            dims.append(dims_list[ix])  # call dimensions in order: x, y, z, q
```
This is confusing to me. You need to assert `len(dims_list) >= len(data_shape)` ? Is it true?

tkeller12 replied:
If I cannot determine the coords from the acquisition parameters file, this section of code will create coords based on the index.

ylin00 replied:
> If I cannot determine the coords from the acquisition parameters file, this section of code will create coords based on the index.

My question is what if `len(data_shape) > len(dims_list)` ?

ylin00 commented on dnplab/dnpImport/prospa.py:
```diff
@@ -239,3 +228,62 @@ def import_csv(path, return_raw=False, is_complex=True):
+
+
+def prospa_coords(attrs, data_shape):
+    """Generate coords from prospa acquisition parameters"""
```
Need docstring for `attrs` and `data_shape`. 
Meanwhile, it is assumed that `attrs["nrPnts"]` and `attrs["dwellTime"]` are `integer` and `float` respectively. If somehow one of them becomes, say `string`, then it will raise some confusing Exceptions. 

Ideally, there needs to be a safeguard asserting that. And better write a test for that.

tkeller12 committed:
Fixed issue with 2d data import in prospa

Using F-ordering instead of C-ordering when reshaping multi-dimensional prospa data

tkeller12 committed:
Adding split function to nddata

Adding split function to split dimension into an additional dimension

tkeller12 committed:
Update dnpNMR.py

Updating how zero_fill_factor is handled in case that it is given as zero or less

tkeller12 committed:
Reformatted to PEP8

thcasey3 reviewed (approved):
I approve while seconding @ylin00 comments

tkeller12 committed:
Update dnplab/dnpImport/prospa.py

Co-authored-by: Yanxian Lin <yanxian@umail.ucsb.edu>

tkeller12 committed:
Reformatting to PEP8

tkeller12 committed:
Update prospa.py

Updated docstring for prospa_coords function
